packages/clarity-js/src/layout/region.ts

```diff
@@ -72,20 +76,25 @@ function handler(entries: IntersectionObserverEntry[]): void {
     for (let entry of entries) {
         let target = entry.target;
         let rect = entry.boundingClientRect;
-
+        let overlap = entry.intersectionRect;
+        let viewport = entry.rootBounds;
         // Only capture regions that have non-zero area to avoid tracking and sending regions
         // that cannot ever be seen by the user. In some cases, websites will have a multiple copy of the same region
         // like search box - one for desktop, and another for mobile. In those cases, CSS media queries determine which one should be visible.
         // Also, if these regions ever become non-zero area (through AJAX, user action or orientation change) - we will automatically start monitoring them from that point onwards
-        if (regionMap.has(target) && rect.width > 0 && rect.height > 0) {
+        if (regionMap.has(target) && rect.width > 0 && rect.height > 0 && viewport.width > 0 && viewport.height > 0) {
             let id = target ? dom.getId(target) : null;
-            let visible = entry.isIntersecting ? BooleanFlag.True : BooleanFlag.False;
-            let d = { id, visible, region: regionMap.get(target) };
-            if (id) {
-                regions[id] = d;
-                updates.push(id);
-                data.push(d);
-            } else { queue.push({node: target, data: d}); }
+            let area = overlap ? (overlap.width * overlap.height * 1.0) / (viewport.width * viewport.height) : 0;
```

Comment:
>area [](start = 16, length = 4)

Minor: Rename to viewportRatio or similar as it's not an area.

Reply:
Done